packages/fxa-settings/src/components/App/index.test.tsx

```diff
@@ -124,16 +159,143 @@ describe('glean', () => {
+  it('when initial metrics query is loading', async () => {
+    (useInitialMetricsQueryState as jest.Mock).mockReturnValueOnce({
+      loading: true,
+    });
+    (useLocalSignedInQueryState as jest.Mock).mockReturnValueOnce({});
+
+    await act(async () => {
+      renderWithLocalizationProvider(
+        <AppContext.Provider
+          value={{ ...mockAppContext(), ...createAppContext() }}
+        >
+          <App flowQueryParams={updatedFlowQueryParams} />
+        </AppContext.Provider>
+      );
+    });
+
+    expect(screen.getByLabelText('Loading…')).toBeInTheDocument();
+  });
+
+  it('when signed in status has not been set yet', async () => {
+    (useInitialMetricsQueryState as jest.Mock).mockReturnValueOnce({
+      loading: false,
+    });
+    (useLocalSignedInQueryState as jest.Mock).mockReturnValueOnce({
+      data: undefined,
+    });
+
+    await act(async () => {
+      renderWithLocalizationProvider(
+        <AppContext.Provider
+          value={{ ...mockAppContext(), ...createAppContext() }}
+        >
+          <App flowQueryParams={updatedFlowQueryParams} />
+        </AppContext.Provider>
+      );
+    });
+
+    expect(screen.getByLabelText('Loading…')).toBeInTheDocument();
+  });
+});
+
+describe('SettingsRoutes', () => {
+  let hardNavigateToContentServerSpy: jest.SpyInstance;
+  const settingsPath = '/settings';
+  jest.mock('@reach/router', () => ({
+    ...jest.requireActual('@reach/router'),
+    useLocation: () => {
+      return {
+        pathname: settingsPath,
+      };
+    },
+  }));
+
+  beforeEach(() => {
+    hardNavigateToContentServerSpy = jest
+      .spyOn(utils, 'hardNavigateToContentServer')
+      .mockImplementation(() => {});
+    (useInitialMetricsQueryState as jest.Mock).mockReturnValue({
+      loading: false,
+    });
+    (useIntegration as jest.Mock).mockReturnValue({
+      getServiceName: jest.fn(),
+    });
+  });
+  afterEach(() => {
+    hardNavigateToContentServerSpy.mockRestore();
+  });
+  it('redirects to /signin if isSignedIn is false', async () => {
+    (useLocalSignedInQueryState as jest.Mock).mockReturnValueOnce({
+      data: { isSignedIn: false },
+    });
+
+    let navigateResult: Promise<void>;
+    await act(async () => {
+      const {
+        history: { navigate },
+      } = renderWithRouter(
+        <AppContext.Provider
+          value={{ ...mockAppContext(), ...createAppContext() }}
+        >
+          <App flowQueryParams={updatedFlowQueryParams} />
+        </AppContext.Provider>,
+        { route: '/' }
+      );
+      navigateResult = navigate(settingsPath);
+    });
+
+    await act(() => navigateResult);
+
+    await waitFor(() => {
+      expect(hardNavigateToContentServerSpy).toHaveBeenCalledWith(
+        `/signin?redirect_to=${encodeURIComponent(settingsPath)}`
+      );
+    });
+  });
+  it.skip('does not redirect if isSignedIn is true', async () => {
+    // TODO before merge - why does this complain with:
```

Comment:
LMK if you see what the heck is going on, otherwise I'll look at this with fresh eyes Monday.